# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/ipc/MMPrinter.h
# Commit: 3b11a30f4146
# Full Hash: 3b11a30f41465b2f5de85bc5ff403d30795f6ca3
# Author: Tom Ritter <tom@mozilla.com>
# Date: 2024-03-23 09:29:17
# Description:
#   Bug 1886922: Do not dereference a Maybe if it is not Some r=mccr8
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D205409
# ==============================================================================

diff -r 16211a33b193 -r 3b11a30f4146 dom/ipc/MMPrinter.h
--- a/dom/ipc/MMPrinter.h	Fri Mar 22 15:13:58 2024 +0000
+++ b/dom/ipc/MMPrinter.h	Fri Mar 22 15:32:39 2024 +0000
@@ -7,6 +7,7 @@
 #ifndef MMPrinter_h
 #define MMPrinter_h
 
+#include "mozilla/Maybe.h"
 #include "mozilla/dom/DOMTypes.h"
 #include "nsString.h"
 
@@ -17,24 +18,40 @@
   static void Print(char const* aLocation, const nsAString& aMsg,
                     ClonedMessageData const& aData) {
     if (MOZ_UNLIKELY(MOZ_LOG_TEST(MMPrinter::sMMLog, LogLevel::Debug))) {
-      MMPrinter::PrintImpl(aLocation, aMsg, aData);
+      Maybe<uint64_t> msgId = MMPrinter::PrintHeader(aLocation, aMsg);
+      if (!msgId.isSome()) {
+        return;
+      }
+      MMPrinter::PrintData(*msgId, aData);
     }
   }
 
   static void Print(char const* aLocation, const nsACString& aActorName,
                     const nsAString& aMessageName,
-                    ClonedMessageData const& aData) {
+                    const Maybe<ClonedMessageData>& aData) {
     if (MOZ_UNLIKELY(MOZ_LOG_TEST(MMPrinter::sMMLog, LogLevel::Debug))) {
-      MMPrinter::PrintImpl(
+      Maybe<uint64_t> msgId = MMPrinter::PrintHeader(
           aLocation,
-          NS_ConvertUTF8toUTF16(aActorName + " - "_ns) + aMessageName, aData);
+          NS_ConvertUTF8toUTF16(aActorName + " - "_ns) + aMessageName);
+
+      if (!msgId.isSome()) {
+        return;
+      }
+
+      if (aData.isSome()) {
+        MMPrinter::PrintData(*msgId, *aData);
+      } else {
+        MMPrinter::PrintNoData(*msgId);
+      }
     }
   }
 
  private:
   static LazyLogModule sMMLog;
-  static void PrintImpl(char const* aLocation, const nsAString& aMsg,
-                        ClonedMessageData const& aData);
+  static Maybe<uint64_t> PrintHeader(char const* aLocation,
+                                     const nsAString& aMsg);
+  static void PrintNoData(uint64_t aMsgId);
+  static void PrintData(uint64_t aMsgId, ClonedMessageData const& aData);
 };
 
 }  // namespace mozilla::dom